src/detectors/FHCAL/FHCAL.cc

```diff
@@ -3,60 +3,208 @@
+            {
+              .energyWeight = "log",
+              .moduleDimZName = "",
+              .sampFrac = 1.0,
+              .logWeightBase = 3.6,
+              .depthCorrection = 0.0,
+              .enableEtaBounds = false
+            },
+            app   // TODO: Remove me once fixed
+          )
+        );
+
+        app->Add(
+          new JChainMultifactoryGeneratorT<Cluster_factory_HcalEndcapPClusters>(
+             "HcalEndcapPClusters",
+            {"HcalEndcapPIslandProtoClusters",  // edm4eic::ProtoClusterCollection
+             "HcalEndcapPHits"},                // edm4hep::SimCalorimeterHitCollection
+            {"HcalEndcapPClusters",             // edm4eic::Cluster
+             "HcalEndcapPClusterAssociations"}, // edm4eic::MCRecoClusterParticleAssociation
+            {
+              .energyWeight = "log",
+              .moduleDimZName = "",
+              .sampFrac = 0.033,
+              .logWeightBase = 6.2,
+              .depthCorrection = 0.0,
+              .enableEtaBounds = false,
+            },
+            app   // TODO: Remove me once fixed
+          )
+        );
 
         app->Add(new JFactoryGeneratorT<RawCalorimeterHit_factory_HcalEndcapPInsertRawHits>());
         app->Add(new JFactoryGeneratorT<CalorimeterHit_factory_HcalEndcapPInsertRecHits>());
         app->Add(new JFactoryGeneratorT<CalorimeterHit_factory_HcalEndcapPInsertMergedHits>());
         app->Add(new JFactoryGeneratorT<ProtoCluster_factory_HcalEndcapPInsertTruthProtoClusters>());
         app->Add(new JFactoryGeneratorT<ProtoCluster_factory_HcalEndcapPInsertIslandProtoClusters>());
-        app->Add(new JFactoryGeneratorT<Cluster_factory_HcalEndcapPInsertClusters>());
-        app->Add(new JFactoryGeneratorT<Cluster_factory_HcalEndcapPInsertTruthClusters>());
+
+        app->Add(
+          new JChainMultifactoryGeneratorT<Cluster_factory_HcalEndcapPTruthClusters>(
+             "HcalEndcapPInsertTruthClusters",
+            {"HcalEndcapPInsertTruthProtoClusters",        // edm4eic::ProtoClusterCollection
+             "HcalEndcapPInsertHits"},                     // edm4hep::SimCalorimeterHitCollection
+            {"HcalEndcapPInsertTruthClusters",             // edm4eic::Cluster
+             "HcalEndcapPInsertTruthClusterAssociations"}, // edm4eic::MCRecoClusterParticleAssociation
+            {
+              .energyWeight = "log",
+              .moduleDimZName = "",
+              .sampFrac = 1.0,
+              .logWeightBase = 3.6,
+              .depthCorrection = 0.0,
+              .enableEtaBounds = true
+            },
+            app   // TODO: Remove me once fixed
+          )
+        );
+
+        app->Add(
+          new JChainMultifactoryGeneratorT<Cluster_factory_HcalEndcapPInsertClusters>(
+             "HcalEndcapPInsertClusters",
+            {"HcalEndcapPInsertIslandProtoClusters",  // edm4eic::ProtoClusterCollection
```

Comment:
```suggestion
            {"HcalEndcapPInsertProtoClusters",  // edm4eic::ProtoClusterCollection
```
as in https://github.com/eic/EICrecon/pull/666/files#diff-105ac13e3fd40754f5377b992a6630adef70b348cf900618fb4477c8a89fdf51L46

Reply:
This doesn't seem right because (on `main`):
https://github.com/eic/EICrecon/blob/dedbad46a5fbe4d043af2b9f0e0e13b4a84ab95d/src/detectors/FHCAL/ProtoCluster_factory_HcalEndcapPInsertIslandProtoClusters.h#L29